x-pack/plugins/enterprise_search/public/applications/app_search/components/result_settings/sample_response/sample_response_logic.test.ts

```diff
@@ -0,0 +1,238 @@
+          id: { raw: 'foo' },
+        });
+      });
+
+      it('calls getSearchResultsSuccess with a "No Results." message if there are no results', async () => {
+        mount();
+        jest.spyOn(SampleResponseLogic.actions, 'getSearchResultsSuccess');
+
+        http.post.mockReturnValue(
+          Promise.resolve({
+            results: [],
+          })
+        );
+
+        SampleResponseLogic.actions.getSearchResults('foo', { foo: { raw: true } });
+        jest.runAllTimers();
+        await nextTick();
+
+        expect(SampleResponseLogic.actions.getSearchResultsSuccess).toHaveBeenCalledWith(
+          'No results.'
+        );
+      });
+
+      it('handles 500 errors by setting a generic error response and showing a flash message error', async () => {
+        mount();
+        jest.spyOn(SampleResponseLogic.actions, 'getSearchResultsFailure');
+
+        const error = {
+          response: {
+            status: 500,
+          },
+        };
+
+        http.post.mockReturnValueOnce(Promise.reject(error));
+
+        SampleResponseLogic.actions.getSearchResults('foo', { foo: { raw: true } });
+        jest.runAllTimers();
+        await nextTick();
+
+        expect(flashAPIErrors).toHaveBeenCalledWith(error);
+        expect(SampleResponseLogic.actions.getSearchResultsFailure).toHaveBeenCalledWith(
+          'An error occured.'
+        );
+      });
+
+      it('handles 400 errors by setting the response message, but does not show a flash error message', async () => {
+        mount();
+        jest.spyOn(SampleResponseLogic.actions, 'getSearchResultsFailure');
+
+        http.post.mockReturnValueOnce(
+          Promise.reject({
+            response: {
+              status: 400,
+            },
+            body: {
+              message: 'A validation error occurred.',
+            },
+          })
+        );
+
+        SampleResponseLogic.actions.getSearchResults('foo', { foo: { raw: true } });
+        jest.runAllTimers();
+        await nextTick();
+
+        expect(SampleResponseLogic.actions.getSearchResultsFailure).toHaveBeenCalledWith(
+          'A validation error occurred.'
+        );
+      });
+
+      it('sets a generic message on a 400 error if no custom message is provided in the response', async () => {
+        mount();
+        jest.spyOn(SampleResponseLogic.actions, 'getSearchResultsFailure');
+
+        http.post.mockReturnValueOnce(
+          Promise.reject({
+            response: {
+              status: 400,
+            },
+          })
+        );
+
+        SampleResponseLogic.actions.getSearchResults('foo', { foo: { raw: true } });
+        jest.runAllTimers();
+        await nextTick();
+
+        expect(SampleResponseLogic.actions.getSearchResultsFailure).toHaveBeenCalledWith(
+          'An error occured.'
+        );
+      });
+
+      it('does nothing if an empty object is passed for the resultFields parameter', async () => {
```

Comment:
Just for my edification, what's the real world scenario where this would be the case? When an engine has no documents?

I tried testing the current view on an empty engine and the entire UI looked a bit odd. I'm kinda wondering if there are no fields, if we should just return early on the entire view with a "You have no documents/fields/schema" type empty prompt and not bother showing the sample response.

Reply:
I don't know, frankly. This happened when I ran the code, so I put a check in for it. I didn't dig deeper. I will though.

There is an additional empty state I haven't implemented yet. I'll do this in a subsequent PR:
<img width="2221" alt="empty" src="https://user-images.githubusercontent.com/1427475/113589920-fa17e080-95ff-11eb-9a69-451280fa46ed.png">

Reply:
Right yeah sorry for being unclear -  I'm suggesting changing the Kibana UI from the screenshotted standalone UI to hide the sample response column entirely if the engine has no schema/fields (i.e., an early return EuiEmptyPrompt). That way we don't run into the need for this check (although to clarify I'm also fine having it just in case).